src/components/sections/contest/GithubFileSelect.tsx

```diff
@@ -0,0 +1,122 @@
+'use client'
+
+import {useState} from 'react'
+import {ArrowLeft} from 'lucide-react'
+import {signIn, useSession} from 'next-auth/react'
+
+import {Button} from '@/components/ui/Button'
+import {
+  useGetPublicRepos,
+  useGetRepoFiles,
+} from '@/lib/queries/github/getGitHub'
+import {
+  GetPublicReposResponse,
+  GetRepoFilesParams,
+  GetRepoFilesResponse,
+} from '@/server/actions/github/getGithub'
+import {Checkbox} from '@/components/ui/Checkbox'
+import {ScrollArea} from '@/components/ui/ScrollArea'
+
+const GithubFileSelect = () => {
+  const session = useSession()
+
+  const [selectedRepo, setSelectedRepo] = useState<GetRepoFilesParams | null>(
+    null,
+  )
+  const {data: publicReposData, isLoading: publicReposIsLoading} =
+    useGetPublicRepos()
+  const {data: repoFilesData, isLoading: repoFilesIsLoading} = useGetRepoFiles(
+    selectedRepo ?? {defaultBranch: '', owner: '', repo: ''},
+    {enabled: selectedRepo !== null},
+  )
+
+  if (session.data?.user.provider !== 'github') {
+    return (
+      <Button
+        onClick={() => signIn('github')}
```

Comment:
Is this working for you? I'm getting `OAuthAccountNotLinked` error, when I'm signed in with a different provider and it asks me to sign in with github. I can investigate it further on Monday 🤔

Reply:
I can't seem to replicate the issue by using a different email address 🤔

Reply:
After resetting DB the Bad credentials error went away, hopefully it was not because of some time-limited token 🤷 The `OAuthAccountNotLinked` sort of makes sense I guess, although it would be nice to make sure user sign in with the Github account that's connected to the same email address, but I am not sure if that's possible with next auth 🤔